modules/cnf-configuring-node-groups-for-the-numaresourcesoperator.adoc

```diff
@@ -0,0 +1,73 @@
+// Module included in the following assemblies:
+//
+// *scalability_and_performance/cnf-numa-aware-scheduling.adoc
+
+:_content-type: PROCEDURE
+
+[id="cnf-configuring-node-groups-for-the-numaresourcesoperator_{context}"]
+= Optional: Configuring polling operations for NUMA resources updates
+
+The daemons controlled by the NUMA Resources Operator in their `nodeGroup` poll resources to retrieve updates about available NUMA resources. You can fine-tune polling operations for these daemons by configuring the `spec.nodeGroups` specification in the `NUMAResourcesOperator` custom resource (CR). This provides advanced control of polling operations. Configure these specifications to improve scheduling behaviour and troubleshoot suboptimal scheduling decisions. 
+
+The configuration options are the following:
+
+* `infoRefreshMode`: Determines the trigger condition for polling the kublet. The NUMA Resources Operator reports the resulting information to the API server. 
+* `infoRefreshPeriod`: Determines the duration between polling updates.
+* `podsFingerprinting`: Determines if point-in-time information for the current set of pods running on a node is exposed in polling updates.
++
+[NOTE]
+====
+`podsFingerprinting` is enabled by default. `podsFingerprinting` is a requirement for the `cacheResyncPeriod` specification in the `NUMAResourcesScheduler` CR. The `cacheResyncPeriod` specification helps to report more exact resource availability by monitoring pending resources on nodes.
+====
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Log in as a user with `cluster-admin` privileges.
+* Install the NUMA Resources Operator.
+
+.Procedure
+
+* Configure the `spec.nodeGroups` specification in your `NUMAResourcesOperator` CR:
++
+[source,yaml]
+----
+apiVersion: nodetopology.openshift.io/v1
+kind: NUMAResourcesOperator
+metadata:
+  name: numaresourcesoperator
+spec:
+  nodeGroups:
+  - config:
+      infoRefreshMode: Periodic <1>
+      infoRefreshPeriod: 10s <2>
+      podsFingerprinting: Enabled <3>
+    name: worker
+----
+<1> Valid values are `Periodic`, `Events`, `PeriodicAndEvents`. Use `Periodic` to poll the kublet at intervals that you define in `infoRefreshPeriod`. Use `Events` to poll the kublet at every pod lifecycle event. Use `PeriodicAndEvents` to enable both methods.
+<2> Define the polling interval for `Periodic` or `PeriodicAndEvents` refresh modes. The field is ignored if the refresh mode is `Events`.
+<3> Valid values are `Enabled` or `Disabled`. Setting to `Enabled` is a requirement for the `cacheResyncPeriod` specification in the `NUMAResourcesScheduler`.
+
+.Verification
+
+. After you deploy the NUMA Resources Operator, verify that the node group configurations were applied by running the following command:
++
+[source,terminal]
+----
+$ oc get numaresop numaresourcesoperator -o json | jq '.status'
+----
++
+.Example output
+[source,terminal]
+----
+      ...
+
+        "config": {
+        "infoRefreshMode": "Periodic",
```

Comment:
I've managed to exec cmd above:
`oc get numaresop numaresourcesoperator -o json | jq '.status'`
My output is a bit different:
      "config": {
        "infoRefreshMode": **"PeriodicAndEvents",**
        "infoRefreshPeriod": "10s",
        "podsFingerprinting": "Enabled"
      },
      "name": "worker"

Reply:
numaresources operator defaults: PeriodicAndEvents
recommended configuration: Periodic

We will evaluate if we can change the defaults, but is not easy.

Reply:
> We will evaluate if we can change the defaults, but is not easy.

We can't. So the manual configuration part will need to include this bit. The user needs anyway to specify the `NodeGroup` (or in general is required to send configuration to the operator). so it should not be a big deal

Reply:
I think the configuration of the NUMAResourcesOperator CR is detailed in this section. And the example output is a reflection of that configuration. So I don't think any changes are required? or I could add a line to say the default is PeriodicAndEvents?

Reply:
@rohennes I agree with you. If the example output reflects the expected outcome of the provided CR, no changes are needed.